### PR TITLE
Allow unstructed schema for config in ProxyDefaults

### PR DESCRIPTION
**Pull request by ishustava**
Please see hashicorp/consul-k8s#495 for details about why this change is needed

Changes proposed in this PR:
- Update CRD schema for ProxyDefaults based on new generated code from the above consul-k8s PR
- Update acceptance test for proxy defaults to set all possible fields. For the `config` field, we're not setting all acceptable values but I think it's OK since we're trying to check that at least one is accepted by Consul.

How I've tested this PR:
- ran acceptance tests

How I expect reviewers to test this PR:
- code review

Checklist:
- [ ] Bats tests added
- [x] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

**Review by thisisnotashwin — APPROVED** [review]
It looks like such a harmless change 😂 Nicely done!!

File: test/acceptance/tests/controller/controller_test.go [hunk 1]
```diff
@@ -92,6 +92,10 @@ func TestController(t *testing.T) {
 					proxyDefaultEntry, ok := entry.(*api.ProxyConfigEntry)
 					require.True(r, ok, "could not cast to ProxyConfigEntry")
 					require.Equal(r, api.MeshGatewayModeLocal, proxyDefaultEntry.MeshGateway.Mode)
+					require.Equal(r, "tcp", proxyDefaultEntry.Config["protocol"])
```
Comment (lkysow) [inline test/acceptance/tests/controller/controller_test.go:95]: let's also test `foo` and `member` for completeness. Or we can remove those from the fixture. I like testing multiple types, so maybe bool, array, map, string?

Comment (ishustava) [reply test/acceptance/tests/controller/controller_test.go:95]: Hmm, good call. I think I've made an assumption that these are going to be ignored by consul, but they're not!

Comment (ishustava) [reply test/acceptance/tests/controller/controller_test.go:95]: done!

**Review by lkysow — APPROVED** [review]

**Review by ndhanushkodi — APPROVED** [review]
Nice investigation and catch! Looks great!!

**Pull request merged**